### PR TITLE
Fix RealNodeData subscriptable

### LINKED ISSUE #7971: TypeError: 'RealNodeData' object is not subscriptable
When running poly_example (on LSF-cluster - maybe not important) getting the following error:
```sh
  File "kenv/root/lib64/python3.8/site-packages/ert/gui/model/snapshot.py", line 379, in data
    return self._job_data(index, node, role)
  File "kenv/root/lib64/python3.8/site-packages/ert/gui/model/snapshot.py", line 462, in _job_data
    return real.data[REAL_JOB_STATUS_AGGREGATED][node.id_]
TypeError: 'RealNodeData' object is not subscriptable

```

### PR DESCRIPTION
**Pull request by andreas-el**
**Issue**
Resolves #7971 

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

**Review by xjules — APPROVED** [review]
Nice @andreas-el ! :rocket:

**Comment by berland** [comment]
This smells like some code not covered by tests. Could these line be identified and an issue for writing a test be written?

**Comment by andreas-el** [comment]
> This smells like some code not covered by tests. Could these line be identified and an issue for writing a test be written?

I am looking into this, yes. I found several accounts of type hinting mismatching the data inserted, which isn't exactly great.

**Comment by berland** [comment]
Very good Job. Go merge.

File: src/ert/gui/model/snapshot.py [hunk 1]
```diff
@@ -232,13 +230,13 @@ def _add_partial_snapshot(self, partial: PartialSnapshot, iter_: int):
                 if job.index:
                     job_node.data.index = job.index
                 if job.current_memory_usage:
-                    current_memory_usage = float(job.current_memory_usage)
-                    job_node.data.current_memory_usage = current_memory_usage
+                    current_memory_usage = int(float(job.current_memory_usage))
```
Comment (xjules) [inline src/ert/gui/model/snapshot.py:233]: do we need variable `current_memory_usage` here at all?

Comment (andreas-el) [reply src/ert/gui/model/snapshot.py:233]: We could instead do;

```
real_node.data.current_memory_usage = int(float(job.current_memory_usage))
self.root.data.current_memory_usage = int(float(job.current_memory_usage))
```

Comment (andreas-el) [reply src/ert/gui/model/snapshot.py]: I've updated the codebase accordingly.

File: tests/unit_tests/gui/model/test_snapshot.py [hunk 2]
```diff
@@ -39,22 +47,107 @@ def test_realization_sort_order(full_snapshot):
...
+        assert colors[i].name() == color_list[i]
+
+        # verify forward_model coloring and status texts
+        status = model.data(model.index(i, 2, first_real), Qt.DisplayRole)
+        color = model.data(model.index(i, 2, first_real), Qt.BackgroundRole)
+        assert status == status_list[i]
+        assert color.name() == color_list[i]
+
+
+def test_realization_queue_wait_color_if_none_running(waiting_snapshot):
```
Comment (xjules) [inline tests/unit_tests/gui/model/test_snapshot.py:99]: I thought to be more explicit with the function name, but not sure tbh `test_that_realization_has_wait_color_when_none_is_running`? :)

File: tests/unit_tests/gui/model/test_snapshot.py [hunk 3]
```diff
@@ -39,22 +47,107 @@ def test_realization_sort_order(full_snapshot):
...
+        assert colors[i].name() == color_list[i]
+
+        # verify forward_model coloring and status texts
+        status = model.data(model.index(i, 2, first_real), Qt.DisplayRole)
+        color = model.data(model.index(i, 2, first_real), Qt.BackgroundRole)
+        assert status == status_list[i]
+        assert color.name() == color_list[i]
+
+
+def test_realization_queue_not_wait_color_if_any_running(waiting_snapshot):
```
Comment (xjules) [inline tests/unit_tests/gui/model/test_snapshot.py:127]: Can this test function be written into one using `pytest.parametrize(..)` instead?

**Review by xjules — CHANGES_REQUESTED** [review]
Looks good, just had a small comment on utilizing parametrization instead of having two test functions.

File: tests/unit_tests/gui/model/test_snapshot.py (hunk 3 above)
Comment (andreas-el) [reply tests/unit_tests/gui/model/test_snapshot.py]: Updated with pytest.parametrize-test.

File: tests/unit_tests/gui/model/test_snapshot.py [hunk 4]
```diff
@@ -39,22 +48,89 @@ def test_realization_sort_order(full_snapshot):
...
         iter_index = model.index(i, 0, model.index(0, 0, QModelIndex()))
 
-        assert str(i) == iter_index.internalPointer().id_, print(
+        assert str(i) == str(iter_index.internalPointer().id_), print(
             i, iter_index.internalPointer()
         )
 
 
-def test_realization_job_hint(full_snapshot):
+def test_realization_forward_model_colors(full_snapshot):
```
Comment (xjules) [inline tests/unit_tests/gui/model/test_snapshot.py:56]: `test_that_realization_state_matches_display_color`

File: tests/unit_tests/gui/model/test_snapshot.py [hunk 5]
```diff
@@ -39,22 +48,89 @@ def test_realization_sort_order(full_snapshot):
...
+            [QColor(*COLOR_FINISHED).name(), QColor(*COLOR_WAITING).name()],
+        ),
+        (
+            FORWARD_MODEL_STATE_RUNNING,
+            [FORWARD_MODEL_STATE_RUNNING, FORWARD_MODEL_STATE_START],
+            [QColor(*COLOR_RUNNING).name(), QColor(*COLOR_PENDING).name()],
+        ),
+    ],
+)
+def test_that_realization_use_wait_when_none_is_running(
```
Comment (xjules) [inline tests/unit_tests/gui/model/test_snapshot.py:115]: `test_display_color_changes_when_realization_state_is_not_running`

**Review by xjules — APPROVED** [review]
Wonderfull job @andreas-el ! :rocket:

**Pull request merged**